spec/key_manager_spec.rb

```diff
@@ -5,76 +5,80 @@
     it 'creates a config object' do
-      config = KeyManager.config_from_repo(repo_name, tx_config)
+      config = KeyManager.config_from_repo(repo_name)
       expect(config).to be_a(Txgh::Config)
     end
 
     it 'creates a config object that contains both project and repo configs' do
-      config = KeyManager.config_from_repo(repo_name, tx_config)
+      config = KeyManager.config_from_repo(repo_name)
       expect(config.project_config).to eq(project_config)
       expect(config.repo_config).to eq(repo_config)
-      expect(config.tx_config).to be_a(TxConfig)
-      expect(config.tx_config.resources.first.project_slug).to eq(project_name)
-    end
-
-    it 'loads tx config from the given file if not explicitly passed in' do
-      path = 'path/to/tx_config'
-      project_config.merge!('tx_config' => path)
-      expect(TxConfig).to receive(:load_file).with(path).and_return(:tx_config)
-      config = KeyManager.config_from_repo(repo_name)
-      expect(config.tx_config).to eq(:tx_config)
     end
   end
 
   describe '.config_from' do
     it 'creates a config object' do
-      config = KeyManager.config_from(project_name, repo_name, tx_config)
+      config = KeyManager.config_from(project_name, repo_name)
       expect(config).to be_a(Txgh::Config)
     end
 
     it 'creates a config object that contains both project and repo configs' do
-      config = KeyManager.config_from(project_name, repo_name, tx_config)
+      config = KeyManager.config_from(project_name, repo_name)
       expect(config.project_config).to eq(project_config)
       expect(config.repo_config).to eq(repo_config)
-      expect(config.tx_config).to be_a(TxConfig)
-      expect(config.tx_config.resources.first.project_slug).to eq(project_name)
     end
+  end
+
+  describe '.tx_config' do
+    let(:config) { KeyManager.config_from(project_name, repo_name) }
+    let(:project) { config.transifex_project }
+    let(:repo) { config.github_repo }
 
-    it 'loads tx config from the given file if not explicitly passed in' do
-      path = 'path/to/tx_config'
+    it 'loads tx config from the given file' do
+      path = 'file://path/to/tx_config'
       project_config.merge!('tx_config' => path)
-      expect(TxConfig).to receive(:load_file).with(path).and_return(:tx_config)
-      config = KeyManager.config_from(project_name, repo_name)
-      expect(config.tx_config).to eq(:tx_config)
+      expect(TxConfig).to receive(:load_file).with('path/to/tx_config').and_return(:tx_config)
+      config = KeyManager.tx_config(project, repo)
+      expect(config).to eq(:tx_config)
+    end
+
+    it 'raises an error if asked to load config from a git repository and no ref is given' do
+      project_config.merge!('tx_config' => 'git://./tx.config')
+      expect { KeyManager.tx_config(project, repo) }.to raise_error(TxghError)
+    end
+
+    it 'loads tx config from a git repository' do
+      project_config.merge!('tx_config' => 'git://./tx.config')
```

Comment:
Heh no. It's definitely not accurate to think of these as URIs - I just needed some kind of vocab word to refer to them. These are just URI-ish in that they start with a scheme.